### PR TITLE
Add annotations for few classes

### PR DESCRIPTION
**Pull request by Laumesis**
Add annotations for fields defined in other files

Remove one empty line to get automatical annotations from fields defined in code in same file.

### CODE REVIEW

File: Database/QuestieDB.lua [hunk 1]
```diff
@@ -1,4 +1,33 @@
...
+---@field itemData any
+---@field questDataTBC nil | table
+---@field objectDataTBC nil | table
+---@field npcDataTBC nil | table
+---@field itemDataTBC nil | table
+---@field sortKeys table
+---@field _itemAdapterQueryOrder table @temporary, until we remove the old db funcitons
+---@field _objectAdapterQueryOrder table @temporary, until we remove the old db funcitons
+---@field _questAdapterQueryOrder table @temporary, until we remove the old db funcitons
+---@field _npcAdapterQueryOrder table @temporary, until we remove the old db funcitons
```
Comment (BreakBB) [inline Database/QuestieDB.lua:30]: These should never be really accessed, so we better don't add them to the "official" class definition

**Comment by BreakBB** [comment]
I was also thinking about adding some dedicated `types.lua` file which contains the definition of all important types and don't bloat the already full module files.

File: Database/QuestieDB.lua (hunk 1 above)
Comment (Laumesis) [reply Database/QuestieDB.lua:30]: Well, those still generate code diagnostics warnings.

**Comment by Laumesis** [comment]
> I was also thinking about adding some dedicated `types.lua` file which contains the definition of all important types and don't bloat the already full module files.

Is someone really going to keep that file up to date? 🙄

**Comment by Laumesis** [comment]
> I was also thinking about adding some dedicated `types.lua` file which contains the definition of all important types and don't bloat the already full module files.

Also would it make git merges harder? All code file changes could possibly change tha same file

**Comment by BreakBB** [comment]
There could be multiple type files and I agree that it might be outdated quite quickly.

Personally I am just not a fan of 20 lines of `---@field foo bar` inside some components and adding a dedicated file next to it is way cleaner imo.

**Comment by drejjmit** [comment]
funcitons 😅

**Comment by Laumesis** [comment]
> funcitons 😅

I copy/pasted 😁

btw, no idea how to do `@field` for function and tell its parameters and return type. Fun sides of defining things all over place in zillion files :(

**Comment by Laumesis** [comment]
Opinions now?

In my opinion we shouldn't define stuff to a module/class outside its file.
As a wild idea: otherwise maybe ".h" file for each module/class that tells all fields. `QuestieFrame_h.lua`? And tell things so well that no need to hunt 1h around code what possible stuff the table may contain 😅

File: Database/QuestieDB.lua (hunk 1 above)
Comment (Laumesis) [reply Database/QuestieDB.lua]: I added private to those

File: Database/QuestieDB.lua [hunk 2]
```diff
@@ -1,4 +1,34 @@
...
+---@field npcData any
+---@field questData any
+---@field objectData any
+---@field itemData any
+---@field questDataTBC nil | table
+---@field objectDataTBC nil | table
+---@field npcDataTBC nil | table
+---@field itemDataTBC nil | table
+---@field sortKeys table
+---@field private private table
```
Comment (BreakBB) [inline Database/QuestieDB.lua:27]:
```suggestion
---@field private table
```

File: Database/QuestieDB.lua [hunk 3]
```diff
@@ -1,4 +1,34 @@
...
+---@field objectKeys table
+---@field objectKeysReversed table
+---@field itemCompilerTypes table
+---@field itemCompilerOrder table
+---@field itemKeys table
+---@field itemKeysReversed table
+---@field npcData any
+---@field questData any
+---@field objectData any
+---@field itemData any
```
Comment (BreakBB) [inline Database/QuestieDB.lua:21]: Those should also be `table` I think

File: Database/QuestieDB.lua [hunk 4]
```diff
@@ -1,4 +1,34 @@
...
+---@field questDataTBC nil | table
+---@field objectDataTBC nil | table
+---@field npcDataTBC nil | table
+---@field itemDataTBC nil | table
+---@field sortKeys table
+---@field private table
```
Comment (BreakBB) [inline Database/QuestieDB.lua:27]:
```suggestion
---@field private private table
```

File: Database/QuestieDB.lua (hunk 3 above)
Comment (Laumesis) [reply Database/QuestieDB.lua]: Done in https://github.com/Questie/Questie/pull/3115/commits/d7d473bcf3757d69408e589e2bbe706d07fb4517

**Review by BreakBB — APPROVED** [review]

**Pull request merged**